src/ethereum/prague/fork.py

```diff
@@ -104,53 +109,32 @@ def apply_fork(old: BlockChain) -> BlockChain:
     ----------
     old :
         Previous block chain object.
+    block :
+        Block to apply to `old`.
 
     Returns
     -------
     new : `BlockChain`
         Upgraded block chain object for this hard fork.
     """
-    return old
-
-
-def get_last_256_block_hashes(chain: BlockChain) -> List[Hash32]:
-    """
-    Obtain the list of hashes of the previous 256 blocks in order of
-    increasing block number.
-
-    This function will return less hashes for the first 256 blocks.
-
-    The ``BLOCKHASH`` opcode needs to access the latest hashes on the chain,
-    therefore this function retrieves them.
-
-    Parameters
-    ----------
-    chain :
-        History and current state.
-
-    Returns
-    -------
-    recent_block_hashes : `List[Hash32]`
-        Hashes of the recent 256 blocks in order of increasing block number.
-    """
-    recent_blocks = chain.blocks[-255:]
-    # TODO: This function has not been tested rigorously
-    if len(recent_blocks) == 0:
-        return []
-
-    recent_block_hashes = []
-
-    for block in recent_blocks:
-        prev_block_hash = block.header.parent_hash
-        recent_block_hashes.append(prev_block_hash)
-
-    # We are computing the hash only for the most recent block and not for
-    # the rest of the blocks as they have successors which have the hash of
-    # the current block as parent hash.
-    most_recent_block_hash = keccak256(rlp.encode(recent_blocks[-1].header))
-    recent_block_hashes.append(most_recent_block_hash)
+    # At fork block, store the hashes for earlier blocks.
+    current_block = block
+    for i in range(HISTORY_SERVE_WINDOW - 1):
```

Comment:
Currently `old.blocks` is only 256 blocks long. So this function will fail if we run the transition on a real chain.

I don't think this is a blocker for this PR, but we should open an issue.